source/simple_sub_pub_demo.c

```diff
@@ -334,6 +375,7 @@ static bool prvSubscribeToTopic( MQTTQoS_t xQoS,
      * the callback executes. */
     xApplicationDefinedContext.ulNotificationValue = ulNextSubscribeMessageID;
     xApplicationDefinedContext.xTaskToNotify = xTaskGetCurrentTaskHandle();
+    xApplicationDefinedContext.pArgs = ( void * ) pcTopicFilter;
```

Comment:
If you set `pArgs` as `&xSubscribeArgs`, you not only have the topic filter but its length as well.

Reply:
Yea I think that is a good suggestion. It will be that was extensible for handling multiple subscriptions as well.